### ISSUE TITLE
Zero to JupyterHub, batchspawner style

### ISSUE THREAD

**Issue by mrocklin** [jupyterhub/batchspawner#101]
There are many batch jobqueue systems out there that are currently considering using JupyterHub.  Currently the advice given to them is to talk to groups that have successfully done it in the past.  This may not be accessible to smaller centers or clusters that don't feel quite as comfortable reaching out.  The Zero-to-JupterHub documentation for Kubernetes has been a success in getting novice groups and users (like mysefl) to build productive systems.  I wonder if we've learned enough about how BatchSpawner gets used in practice to provide best practice guidelines

cc @minrk @choldgraf

**Comment by rkdarst** [comment]
Hi,

Have you seen this: https://github.com/jupyterhub/jupyterhub-deploy-hpc
And my own personal project is: https://github.com/AaltoScienceIT/triton-jupyterhub

This is something I wanted to try to do with my personal project, but haven't gotten very far (and haven't worked on the docs in a long time).  Perhaps it is time to start again.  My vision would be: kubernetes deployment is more ephermal, while the batchspawer guide is focused on integrating with existing system and making all the software, storage, etc on that system available to the jupyter notebooks (I'm biased because this was my goal...).

Is anyone else interested in working on this guide?  We could start in wiki format here on github and then see where it goes from there, and I would donate my material as a starting point.

**Comment by mbmilligan** [comment]
I quite agree that this is a good idea. The jupyterhub-deploy-hpc repo was intended to be a home for exactly that type of documentation, but the relevant people (myself included) haven't had time to build it out. I'd certainly be interested in trying to kick off a second attempt at that.

I will note that, while I think the community does have pretty good knowledge about how to build BatchSpawner-based deployments now, there probably won't be a one-size-fits-all approach that makes sense to the degree that z2jh does. Nobody here is starting from zero, since the premise is "so you have a supercomputer..."

That said, we can and definitely should write up some best practices to the extent that we can generalize, and accompany that with worked examples for a few common scenarios.

**Comment by mbmilligan** [comment]
I agree with using wiki pages to initially develop the content. This outline looks like a reasonable approach, and if you have some content that can help kickstart the process I think that would be very helpful to getting momentum going on this.

I've created an initial wiki page here: https://github.com/jupyterhub/jupyterhub-deploy-hpc/wiki

In theory all Github users should be able to edit that, but let me know if that doesn't work.

**Comment by willingc** [comment]
Alternatively, I could give you both push access to the jupyterhub-deploy-hpc repo.

**Comment by mbmilligan** [comment]
@willingc yes, no reason not to give them push access there. But if they want to use the wiki to develop content, that's available too.

**Comment by willingc** [comment]
Gave @mrocklin @rkdarst push privileges on the repo. You should receive invites.

**Comment by guillaumeeb** [comment]
Hi everyone, very interested in this too, as I am currently deploying a jupyterhub instance on our PBS HPC Cluster. Thanks @mrocklin for pointing that to me.

@rkdarst, your outline looks good to me!

**Comment by guillaumeeb** [comment]
Hi,

Is there any advance here? I tried to play with the wiki but, it seems I'm not able to do any modifications.
I've deployed a beta version of Jupyterhub on our System at the end of august, so I may have some things to say.

I would add some points to @rkdarst outiline above (that may already be forseen in other bigger bullets):
- Security : get SSL certificate, reverse proxy, iptables...
- Kernel creation for users or admin.
- Possibly: deploy other language kernel?
